dashboard/src/components/billing/UsageUpgradeBanner.tsx

```diff
@@ -0,0 +1,60 @@
+'use client';
+
+import { AlertTriangle, AlertCircle } from 'lucide-react';
+import { Button } from '@/components/ui/button';
+import Link from 'next/link';
+import { use } from 'react';
+import type { UserBillingData } from '@/entities/billing';
+import { isClientFeatureEnabled } from '@/lib/client-feature-flags';
+
+interface UsageUpgradeBannerProps {
+  billingDataPromise: Promise<UserBillingData>;
+}
+
+export default function UsageUpgradeBanner({ billingDataPromise }: UsageUpgradeBannerProps) {
+  if (!isClientFeatureEnabled('enableBilling')) {
+    return null;
+  }
+
+  const billingData = use(billingDataPromise);
```

Comment:
Conditionally rendering hooks is not allowed in React:
```suggestion
  const billingData = use(billingDataPromise);

  if (!isClientFeatureEnabled('enableBilling')) {
    return null;
  }
```

Reply:
I've removed the feature guard inside the component and kept the guard-wrapper :)